itests/fevm_test.go

```diff
@@ -112,3 +95,109 @@ func TestFEVMETH0(t *testing.T) {
+	expectedResult, err := hex.DecodeString("0000000000000000000000000000000000000000000000000000000000000007")
+	require.NoError(t, err)
+	require.Equal(t, result, expectedResult)
+
+	//test the value is 7 via calling the getter
+	result = client.EVM().InvokeContractByFuncName(ctx, fromAddr, storageAddr, "getCounter()", []byte{})
+	require.Equal(t, result, expectedResult)
+
+	//test the value is 0 via calling the getter on the Actor contract
+	result = client.EVM().InvokeContractByFuncName(ctx, fromAddr, actorAddr, "getCounter()", []byte{})
+	expectedResultActor, err := hex.DecodeString("0000000000000000000000000000000000000000000000000000000000000000")
+	require.NoError(t, err)
+	require.Equal(t, result, expectedResultActor)
+}
+
+// TestFEVMRecursiveFuncCall deploys a contract and makes a recursive function calls
+func TestFEVMRecursiveFuncCall(t *testing.T) {
+	ctx, cancel, client := setupFEVMTest(t)
+	defer cancel()
+
+	//install contract Actor
+	filenameActor := "contracts/StackFunc.hex"
+	fromAddr, actorAddr := client.EVM().DeployContractFromFilename(ctx, filenameActor)
+
+	testN := func(n int) func(t *testing.T) {
+		return func(t *testing.T) {
+			inputData := make([]byte, 32)
+			binary.BigEndian.PutUint64(inputData[24:], uint64(n))
+
+			result := client.EVM().InvokeContractByFuncName(ctx, fromAddr, actorAddr, "exec1(uint256)", inputData)
+			require.Equal(t, result, []byte{})
+		}
+	}
+
+	t.Run("n=0", testN(0))
+	t.Run("n=1", testN(1))
+	t.Run("n=20", testN(20))
+	t.Run("n=200", testN(200))
+	t.Run("n=293", testN(293))
+}
+
+// TestFEVMRecursiveActorCall deploys a contract and makes a recursive actor calls
+func TestFEVMRecursiveActorCall(t *testing.T) {
+	ctx, cancel, client := setupFEVMTest(t)
+	defer cancel()
+
+	//install contract Actor
+	filenameActor := "contracts/RecCall.hex"
+	fromAddr, actorAddr := client.EVM().DeployContractFromFilename(ctx, filenameActor)
+
+	testN := func(n, r int) func(t *testing.T) {
+		return func(t *testing.T) {
+			inputData := make([]byte, 32*3)
+			binary.BigEndian.PutUint64(inputData[24:], uint64(n))
+			binary.BigEndian.PutUint64(inputData[32+24:], uint64(n))
+			binary.BigEndian.PutUint64(inputData[32+32+24:], uint64(r))
+
+			result := client.EVM().InvokeContractByFuncName(ctx, fromAddr, actorAddr, "exec1(uint256,uint256,uint256)", inputData)
+			require.Equal(t, result, []byte{})
+		}
+	}
+
+	t.Run("n=0,r=1", testN(0, 1))
+	t.Run("n=1,r=1", testN(1, 1))
+	t.Run("n=20,r=1", testN(20, 1))
+	t.Run("n=200,r=1", testN(200, 1))
+	t.Run("n=251,r=1", testN(251, 1))
+
+	t.Run("n=0,r=10", testN(0, 10))
+	t.Run("n=1,r=10", testN(1, 10))
+	t.Run("n=20,r=10", testN(20, 10))
+	t.Run("n=200,r=10", testN(200, 10))
+	t.Run("n=251,r=10", testN(251, 10))
+
+	t.Run("n=0,r=32", testN(0, 32))
+	t.Run("n=1,r=32", testN(1, 32))
+	t.Run("n=20,r=32", testN(20, 32))
+	t.Run("n=200,r=32", testN(200, 32))
+	t.Run("n=251,r=32", testN(251, 32))
+
+	t.Run("n=0,r=33", testN(0, 33))
```

Comment:
The tests above (with r <= 32) all pass, at 32 (those are recursive actor calls) I'm seeing
```
    evm.go:237: gas used: 102988605
    evm.go:238: 
        	Error Trace:	/home/magik6k/github.com/filecoin-project/go-lotus/itests/evm.go:238
        	            				/home/magik6k/github.com/filecoin-project/go-lotus/itests/fevm_test.go:179
        	Error:      	Should be true
        	Test:       	TestFEVMRecursiveActorCall
        	Messages:   	contract execution failed: 33
```

Which I'm not exactly sure what that means. I'd expect the actor invocation limit to be 1k. Did that change recently?

Reply:
It should be 1k, but we might be hitting a memory limit here?

Reply:
Turns out this was https://github.com/filecoin-project/lotus/issues/10041